ophyd/controls/areadetector/filestore_mixins.py

```diff
@@ -0,0 +1,167 @@
+    from ophyd.controls.areadetector.plugins import HDF5Plugin
+    from ophyd.controls.areadetector.trigger_mixins import SingleTrigger
+    from ophyd.controls.areadetector.filestore_mixins import (
+        FileStoreIterativeWrite)
+
+    class MyPlugin(HDF5Plugin, FileStoreIterativeWrite):
+        pass
+
+    dest = '/tmp'  # in production, use a directory on your system -- not /tmp
+
+    class MyDetector(PerkinElmerDetector, SingleTrigger):  # for example
+        file_plugin = MyPlugin(suffix='HDF1:', write_file_path=dest)
+
+    det = MyDetector(...)
+"""
+
+from __future__ import print_function
+import logging
+import uuid
+import filestore.api as fs
+
+from datetime import datetime
+from collections import defaultdict, OrderedDict
+from itertools import count
+import os
+
+from ..ophydobj import DeviceStatus
+from ..device import GenerateDatumInterface, BlueskyInterface
+
+logger = logging.getLogger(__name__)
+
+
+def new_uid():
+    "uuid4 as a string"
+    return str(uuid.uuid4())
+
+
+def new_short_uid():
+    "uuid4, skipping the last stanza because of AD length restrictions."
+    return '-'.join(new_uid().split('-')[:-1])
+
+
+class FileStoreBase(BlueskyInterface, GenerateDatumInterface):
+    "Base class for FileStore mixin classes"
+    def __init__(self, *args, write_file_path=None, read_file_path=None,
+                 **kwargs):
+        # TODO Can we make these args? Depends on plugin details.
+        if write_file_path is None:
+            raise ValueError("write_file_path is required")
+        self.write_file_path = write_file_path
+        if read_file_path is None:
+            self.read_file_path = write_file_path
+        else:
+            self.read_file_path = read_file_path
+        super().__init__(*args, **kwargs)
+        self._point_counter = None
+        self._locked_key_list = False
+        self._datum_uids = defaultdict(list)
+
+    def stage(self):
+        self._point_counter = count()
+        self._locked_key_list = False
+        self._datum_uids.clear()
+        # Make a filename.
+        self._filename = new_short_uid()
+        date = datetime.now().date()
+        # AD requires a trailing slash, hence the [''] here
+        path = os.path.join(*(date.isoformat().split('-') + ['']))
+        full_write_path = os.path.join(self.write_file_path, path)
+        full_read_path = os.path.join(self.read_file_path, path)
+        self.file_template.put('%s%s_%6.6d.h5')
+        ssigs = [
+            (self.enable, 1),
+            (self.auto_increment, 1),
+            (self.array_counter, 0),
+            (self.file_number, 0),
+            (self.auto_save, 1),
+            (self.num_capture, 0),
+            (self.file_write_mode, 1),  # 'capture' mode -- for AD 1.x
+            (self.file_path, full_write_path),
+            (self.file_name, self._filename),
+            (self.capture, 1),
+        ]
+        self.stage_sigs.extend(ssigs)
+        print('staging fsbase')
```

Comment:
I was about to nitpick about your spelling, but it is a "rouge rogue print" with the syntax colorscheme.